### DISCUSSION TITLE
Allow searchbar manipulation

### DISCUSSION THREAD

**Discussion by ha3** [software-mansion/react-native-screens#1323] [Ideas]
Hello,

As far as I know, there is no way to programmatically focus to or blur from a search bar now. I think it would be very helpful to add this feature. For API, maybe search bar options could accept a ref prop.

Thanks

**Comment by ha3** [comment]
@kacperkapusciak @WoLewicki what do you think about this? Another use case would be to clear the text input. I assume there is no way to achieve that for now?

**Reply by WoLewicki** [reply]
We are always open for PRs with new functionallities :tada: If you would be able to also provide a code example in [TestExample](https://github.com/software-mansion/react-native-screens/tree/main/TestsExample) of how it can be helpful, it would greatly shorten the time needed to test and eventually merge it.

**Comment by henninghall** [comment]
In addition to `focus` and `blur`, a `clear` function would be useful too!

**Comment by patricronge** [comment]
I made a PR #1639 for adding an optional prop for clearing the search input that runs when search button is pressed. Would like to implement for iOS as well if you'd consider this approach. 
Only implemented Android to reduce the scope since it's my first contribution to an open source project.

**Reply by kkafar** [reply]
Thank you! There is also [this amazing PR](https://github.com/software-mansion/react-native-screens/pull/1523). But as I'm explaining there, I'm waiting for few important fixes that will most likely come with `ReactNative@0.71.0`. I'll try to push your ideas after the release.

**Reply by patricronge** [reply]
Ah, didn't notice that PR. That looks super helpful! Thanks for the update

**Comment by felixaa** [comment]
What would the effort be to implement a controlled input? Does UISearchController support this?

**Reply by kkafar** [reply]
Hi @felixaa!
Could you elaborate on what you mean exactly by "controlled input"?

**Reply by felixaa** [reply]
Sorry, let me give some examples

In the case of wanting a default value when configuring the SearchBar initially 
```
{
  headerSearchBarOptions: {
    placeholder: "Search",
    autoFocus: true,
+  value: 'someValue'
  }
}
```

Or a way to set the value 
```
navigation.setOptions({ headerSearchBarOptions: { value: 'someValue' } })
```

I have a use case where I have a "recent searches" list. When pressing one of those list items, I want to put that string into the search header.

**Comment by tboba** [comment]
Hi @ha3, sorry for the late response 🙏 
I believe you can focus or blur from the search bar with the use of the imperative API:

```
const searchBarRef = useRef<SearchBarCommands>(null);

// screenOptions or navigation#setOptions

          ref: searchBarRef,
```

with this, simple do `searchBarRef.current.focus()` or `.blur()` and this should work perfectly.
Cheers 🤞

**Reply by cindy8881** [reply]
Hi
